File: app/src/main/java/ru/henridellal/dialer/RegexQueryResult.java
package ru.henridellal.dialer;

public class RegexQueryResult implements Comparable<RegexQueryResult> {
	public final int position;
	public final int start;
	public final int end;

	public final int id;
	public final String lookupKey;
	public final String name;
	public final String number;
	public int numberStart;
	public int numberEnd;
	
	public RegexQueryResult(int position, int start, int end, int id, String lookupKey, String name, String number) {
		this.position = position;
		this.start = start;
		this.end = end;
		this.id = id;
		this.lookupKey = lookupKey;
		this.name = name;
		this.number = number;
	}
	
	public void setNumberPlace(int numberStart, int numberEnd) {
		this.numberStart = numberStart;
		this.numberEnd = numberEnd;
	}
	
	@Override
	public int compareTo(RegexQueryResult obj) throws NullPointerException, ClassCastException {
		if (null == obj) {
			throw new NullPointerException();
		}
		int result = Integer.compare(this.start, obj.start);
		return (result != 0) ? result : Integer.compare(this.position, obj.position);
	}
}

File: app/src/main/java/ru/henridellal/dialer/RegexQueryResult.kt
package ru.henridellal.dialer

class RegexQueryResult(@JvmField val position: Int, @JvmField val start: Int, @JvmField val end: Int) : Comparable<RegexQueryResult?> {
    @JvmField
	var numberStart = 0
    @JvmField
	var numberEnd = 0
    fun setNumberPlace(numberStart: Int, numberEnd: Int) {
        this.numberStart = numberStart
        this.numberEnd = numberEnd
    }

    @Throws(NullPointerException::class, ClassCastException::class)
    override fun compareTo(obj: RegexQueryResult?): Int {
        if (null == obj) {
            throw NullPointerException()
        }
        val result = start.compareTo(obj.start)
        return if (result != 0) result else position.compareTo(obj.position)
    }
}